src/main/java/com/crumbs/fss/service/RestaurantService.java

```diff
@@ -115,19 +94,26 @@ public Restaurant deleteRestaurant(Long id){
 
         return temp;
     }
+    public Restaurant deleteRestaurantOwnerRestaurant(Long id){
+        Restaurant temp = restaurantRepository.findById(id).orElseThrow(() -> new EntityNotFoundException());
+        restaurantRepository.deleteById(id);
+        locationRepository.deleteById(temp.getLocation().getId());
+        if(menuItemRepository.findById(id).isPresent())
+            menuItemRepository.deleteById(id);
+
+        return temp;
+
+    }
     public Restaurant updateRestaurant(Long id, updateRestaurantDTO updateRestaurantDTO){
 
-        if(restaurantRepository.findById(id).isEmpty())
-            throw new EntityNotFoundException();
+        Restaurant temp = restaurantRepository.findById(id).orElseThrow(() -> new EntityNotFoundException());
 
         if(userDetailRepository.findUserByEmail(updateRestaurantDTO.getEmail())!=null)

```

Comment:
You could have your JPQL return an optional and if the entity isn't present throw an EntityNotFoundException